Experiment fails if discovery found no nodes

```
2023-08-24T14:34:10.453Z        INFO    itn orchestrator        itn_orchestrator/main.go:233    Performing step wait (47)
Starting round 4, 240 min since start
2023-08-24T15:03:59.455Z        INFO    itn orchestrator        itn_orchestrator/main.go:233    Performing step discovery (48)
2023-08-24T15:03:59.704Z        INFO    itn orchestrator        itn_orchestrator/main.go:233    Performing step load-keys (49)
2023-08-24T15:06:34.383Z        ERROR   itn orchestrator        itn_orchestrator/main.go:215    Error resolving params for step 50: couldn't find output participant (step 48, file "") needed for step 50
```

Similar issue:

```
Stopping 30.9% block producers without cleaning
{"step":58,"name":"rest","value":["10.233.71.30:9000","10.233.73.83:9000","10.233.114.254:9000"]}
2023-09-06T03:12:27.323Z        INFO    itn orchestrator        itn_orchestrator/main.go:234    Performing step stop-daemon (59)

2023-09-06T03:12:27.324Z        ERROR   itn orchestrator        itn_orchestrator/main.go:237    Error running step 59: error stoping daemon on 10.233.80.30:9000 (delay 10): Post "http://10.233.80.30:9000/graphql": dial tcp 10.233.80.30:9000: connect: connection refused
```

--- a/src/app/itn_orchestrator/src/discovery.go
+++ b/src/app/itn_orchestrator/src/discovery.go
@@ -32,7 +32,7 @@ type DiscoveryParams struct {
 	Exactly            bool `json:"exactly,omitempty"`
 }
 
-func DiscoverParticipants(config Config, params DiscoveryParams, output func(NodeAddress)) error {
+func discoverParticipantsDo(config Config, params DiscoveryParams, output func(NodeAddress)) error {
 	before := time.Now().Add(-time.Duration(params.OffsetMin) * time.Minute)
 	query := &storage.Query{StartOffset: prefixByTime(before)}
 	log := config.Log
@@ -87,7 +87,24 @@ func DiscoverParticipants(config Config, params DiscoveryParams, output func(Nod
 	if len(cache) != params.Limit && params.Exactly {
 		return errors.New("failed to discover the exact number of nodes")
 	}
+	if len(cache) == 0 {
+		return errors.New("didn't find any participants")
+	}
 	return nil
+}
+
+func DiscoverParticipants(config Config, params DiscoveryParams, output func(NodeAddress)) (err error) {
+	for retryPause := 10; retryPause <= 40; retryPause = retryPause * 2 {
+		err = discoverParticipantsDo(config, params, output)
+		if err == nil {
+			return
+		}
+		if retryPause <= 20 {
+			config.Log.Warnf("Failed to discover participants, retrying in %d minutes: %s", retryPause, err)
+			time.Sleep(time.Duration(retryPause) * time.Minute)
+		}
+	}
+	return
 }
 
 type DiscoveryAction struct{}

--- a/src/app/itn_orchestrator/src/stop.go
+++ b/src/app/itn_orchestrator/src/stop.go
@@ -45,17 +45,13 @@ type StopDaemonParams struct {
 }
 
 func StopDaemon(config Config, params StopDaemonParams) error {
-	errs := []error{}
 	for _, addr := range params.Nodes {
 		resp, err := StopDaemonGql(config, addr, params.Clean, config.StopDaemonDelaySec)
 		if err == nil {
 			config.Log.Infof("stopped daemon on %s: %s", addr, resp)
 		} else {
-			errs = append(errs, err)
+			config.Log.Warnf("failed to stop daemon on %s: %s", addr, err)
 		}
-	}
-	if len(errs) > 0 {
-		return errors.Join(errs...)
 	}
 	return nil
 }

